Fixes #32584 - drop flot charts

ezr-ondrej opened the pull request:
All uses of flot charts have been deprecated and are rendering React
counterparts. We can remove flot charts now :boom:

~~Need to wait for the last deprecation #8518~~

ezr-ondrej committed:
Fixes #32584 - drop flot charts

All uses of flot charts have been deprecated and are rendering React
counterparts. We can remove flot charts now.

tbrisker commented on app/assets/javascripts/charts.js:
```diff
@@ -1,264 +1 @@
-  $('.statistics-bar:visible').flot_bar();
-  $('.statistics-chart:visible').flot_chart();
-}
+tfm.tools.deprecate('charts.js file', 'React chart components', '3.1');
```
Should we just remove this file completely? anyone attempting to use it would fail anyways

ezr-ondrej replied:
Yeah, but they will get this deprecation message to see why it failed, but I guess they can find the info other ways :+1:

tbrisker commented on app/assets/stylesheets/charts.scss:
```diff
@@ -1,157 +0,0 @@
-.stats-well {
-  border: 1px solid #d7d7d7;
-  background-size: 100%;
-  padding: 15px;
-  min-width: 280px;
-  min-height: 100px;
-  margin-bottom: 15px;
-
-  h2,
-  h4 {
-    font-weight: normal;
-  }
-}
```
This class is still used in some places

ezr-ondrej replied:
I hope I've done all that are still needed.

tbrisker commented on app/assets/stylesheets/charts.scss:
```diff
@@ -1,157 +0,0 @@
-  padding-bottom: 40px;
-}
-
-.report-chart {
-  height: 405px;
-}
-
-.host-configuration-chart {
-  width: 240px;
-  margin: 25px auto;
-}
-
-#host-show {
-  .statistics-chart {
-    height: 230px;
-  }
-
-  .chart {
-    height: 250px;
-  }
-}
```
I thinks theses classes as well

ezr-ondrej replied:
These just made the current charts look bad.
I've added commit before the drop that migrates to rendering the React components for Host detail page charts directly and makes them look better at least spacialy.

tbrisker reviewed (changes requested):
some of the css classes are still used in other places, please move them to one of the other stylesheets

ezr-ondrej committed:
Fixes #23665 - Patternfly charts on Host Detail

We are still using deprecated method `flot_chart` to render the Host
detail page charts. It works, but we should migrate to rendering
React components directly.

ezr-ondrej commented:
It now also include direct rendering of Host Detail page charts as separate commit, so we do not have deprecation warnings rendered by core.
I hope I've taken care of all the still usefull styles in charts, but styles are hard, so not sure, but I'd prefer droping and fixing some minor visual issues if those appear.

tbrisker commented on app/views/hosts/_resources.html.erb:
```diff
@@ -1 +1,5 @@
+                                 xAxisLabel: '',
+                                 yAxisLabel: _('Number of Events'),
+                                 data: resources_chart_data(params["range"].to_i.days.ago),
+                                 size: { height: 270 }) %>
```
This (and the other one) used to be 250 previously, for me at least 270 causes the page to overflow.

tbrisker reviewed (commented):
Looks good @ezr-ondrej , great cleanup! one inline comment, other than that it should be good to go. can you please post on discourse notifying plugin maintainers that they should update their plugins and maybe even list the removed functions?

ezr-ondrej commented:
I've created a discourse anouncement: https://community.theforeman.org/t/warning-flot-charts-are-going-away/24318

tbrisker reviewed (approved):
Thanks @ezr-ondrej !
